gbbs/benchmarks/BatchDynamicConnectivity/EulerTourTree/ETTree.h

```diff
@@ -591,6 +591,38 @@ struct ETTree {
         return edge_sum;
         //return edge_sum.first ^ edge_sum.second;
     }
+    
+    template <class KY, class VL, class HH>
+    sequence<sequence<sequence<std::pair<uintE, uintE>>>> batch_sub_tree_sum(sequence<std::pair<uintE,uintE>> v_p,
+            pbbslib::sparse_table<KY, VL, HH> edge_index_table)
+    {
+
+
+        size_t num_queries = v_p.size();
+  
+        sequence<sequence<sequence<std::pair<uintE, uintE>>>> subtree_sums(num_queries);
+
+        parallel_for(0, num_queries, [&](size_t i){
+
+            uintE u = v_p[i].first; 
+            uintE v = v_p[i].second; 
```

Comment:
```suggestion
            auto const [u, v] = v_p[i];
```
this is a nice spot to use [structured bindings](https://www.cppstories.com/2022/structured-bindings/) (if you have c++17 enabled). Using tuples and structured bindings leads to very nice semantics for passing "sum objects" around without creating custom structs everywhere. (Definitely an optional comment in this context, but a practice I'd like you to get in the habit of.)

Reply:
Yes this is more semantically clear.